d3rlpy/algos/qlearning/torch/cql_impl.py

```diff
@@ -75,7 +75,23 @@ def compute_critic_loss(
         conservative_loss = self._compute_conservative_loss(
             batch.observations, batch.actions, batch.next_observations
         )
-        return loss + conservative_loss
+        return loss + conservative_loss, conservative_loss
+
+    @train_api
+    def update_critic(self, batch: TorchMiniBatch) -> np.array:
```

Comment:
This is altering a signature of `update_critic`, which is `def update_critic(self, batch: TorchMiniBatch) -> float:`. Thus, we actually need to refactor these methods first. I'll take the first pass to return `Dict[str, float]`. Then, you can make changes on top of it. Sorry for the inconvenience. I'll let you know when it's done.

Reply:
The interface has been updated in this commit: https://github.com/takuseno/d3rlpy/commit/67723bef91a38bb111aa6480b4d2f5ed7008eb9f . Please check.

Reply:
@takuseno - I'm sorry for the delay! All looks good to me except my one comment on the AWAC d3rlpy/algos/qlearning/awac.py file. Cheers

Reply:
@takuseno is there anything else you need from me at all for the PR? :)

Reply:
@joshuaspear Thanks for the contribution. I've changed the target branch to `refactor_loss` because I've made some new changes to master branch. I'm seeing conflicts between your PR and `refactor_loss` branch. Could you resolve it? Here is an example instruction to merge `refactor_loss` branch to your branch.
```
$ git fetch upstream
$ git checkout master
$ git merge upstream/refactor_loss
```

Reply:
@takuseno no probs - will do

Reply:
@takuseno have merged the branches - I included a couple more data classes for the loss outputs FYI